include/modem/sms.h

```diff
@@ -62,14 +181,28 @@ int sms_register_listener(sms_callback_t listener, void *context);
 /**
  * @brief Unregister a listener.
  *
- * @param handle Handle identifying the listener to unregister.
+ * @details Also unregisters from modem's SMS service if there are
+ * no listeners registered.
+ *
+ * @param[in] handle Handle identifying the listener to unregister.
  */
 void sms_unregister_listener(int handle);
 
 /**
- * @brief Uninitialize the SMS subscriber module.
+ * @brief Send SMS message.
+ *
+ * @details Sending is done with GSM 7bit encoding used to encode textual SMS messages.
+ * SMS-SUBMIT message is specified in 3GPP TS 23.040 Section 9.2.2.2 and data encoding
+ * in 3GPP TS 23.038 Section 4 and Section 6.2.
+ * This function doesn't support sending of 8bit binary data messages or UCS2 encoded text.
+ *
+ * @param[in] number Recipient number.
+ * @param[in] text Text to be sent.
+ *
+ * @retval -EINVAL Invalid parameter.
+ * @return 0 on success, otherwise error code.
  */
-void sms_uninit(void);
+int sms_send_text(char *number, char *text);
```

Comment:
Both `number` and `text` are constant, so `const char *` would be better.

Reply:
Changed